### ISSUE TITLE
Rule proposal: prefer-operator-assignement

### ISSUE THREAD

**Issue by B2o5T** [sindresorhus/eslint-plugin-unicorn#1805]
### Description

if the new string starts with the previous string, prefer operator assignment as less code is written

### Fail

```js
foo = foo + 'baz qux'
```

```js
foo = `${foo} baz ${qux}`
```

### Pass

```js
foo += 'baz qux'
```

```js
foo += ` baz ${qux}`
```

**Comment by sindresorhus** [comment]
> ESLint's [operator-assignment](https://eslint.org/docs/rules/operator-assignment) doesn't catch case with template literal interpolation

Can you open an issue on ESLint about the `operator-assignment` covering this case too?

**Comment by B2o5T** [comment]
I opened, but it was rejected  https://github.com/eslint/eslint/issues/15840#event-6557480224

**Comment by sindresorhus** [comment]
Accepted
